homeassistant/components/twitch/sensor.py

```diff
@@ -51,28 +60,45 @@ def setup_platform(
     """Set up the Twitch platform."""
     channels = config[CONF_CHANNELS]
     client_id = config[CONF_CLIENT_ID]
+    client_secret = config[CONF_CLIENT_SECRET]
     oauth_token = config.get(CONF_TOKEN)
-    client = TwitchClient(client_id, oauth_token)
+    client = Twitch(app_id=client_id, app_secret=client_secret)
```

Comment:
`authenticate_app` is True by default, so this will now try to authenticate here (without any scopes)
And then it's trying again on L69 with scope, it should be deactivated here, or guarded with try/except as this construction can raise